src/main/java/com/andersen/validator/CustomConstraintValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.andersen.validator;
+
+import com.andersen.bus.TicketType;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface CustomConstraintValidator<T> {
```

Comment:
good idea. But in this case this validator deends of Ticket (TicketType), so you can cretae just BusTicketValidator  without this interface

Reply:
> good idea. But in this case this validator deends of Ticket (TicketType), so you can cretae just BusTicketValidator without this interface
Is it because this interface used just by one class BusTicketValidator, that's why we don't need to implement interface in this case, am I right?

Reply:
Basically yes) 
And it just can't be used by others because it is linked to Tickets).
In order to do something like this, you could, for example, do basic validation of string length, id value and something basic.